packages/ember-data/tests/integration/adapter/store-adapter-test.js

```diff
@@ -470,15 +470,13 @@ test("if a created record is marked as invalid by the server, it enters an error
     yehuda.save().then(null, async(function(error) {
       equal(get(yehuda, 'isValid'), false, "the record is invalid");
       ok(get(yehuda, 'errors.name'), "The errors.name property exists");
-
       set(yehuda, 'updatedAt', true);
       equal(get(yehuda, 'isValid'), false, "the record is still invalid");
-
       set(yehuda, 'name', "Brohuda Brokatz");
-
-      equal(get(yehuda, 'isValid'), true, "the record is no longer invalid after changing");
+      //shouldn't we use rollbackAttributes. or perhaps a rollbackAttribute method is needed.
+      //keeping track of changes this is fragile.
```

Comment:
@arenoir I'd like to get a better understanding of what you are suggesting here. Are you saying rollbackAttributes should be responsible for transitioning a model out of the invalid state?